src/kernels/helpers.ts

```diff
@@ -212,13 +211,21 @@ export function getDisplayNameOrNameOfKernelConnection(kernelConnection: KernelC
                     return kernelConnection.kernelSpec.display_name;
                 }
                 const pythonDisplayName = pythonVersion.trim();
-                return kernelConnection.interpreter.envName
-                    ? `${kernelConnection.interpreter.envName} (${pythonDisplayName})`
-                    : pythonDisplayName;
+                const envName = getPythonEnvironmentName(kernelConnection.interpreter);
+                return envName ? `${envName} (${pythonDisplayName})` : pythonDisplayName;
             }
     }
     return oldDisplayName;
 }
+function getPythonEnvironmentName(pythonEnv: PythonEnvironment) {
+    // Sometimes Python extension doesn't detect conda environments correctly (e.g. conda env create without a name).
+    // In such cases the envName is empty, but it has a path.
```

Comment:
Basically I was ending up with `Python 3.6` as the display name for a lot of my test environments.
This ensures we have a better display name (e.g. `test1 (Python 3.6)` which is the standard we've been using when we have an env name). 
In such cases the envname is not available as you activate using the fully qualified path, however you can also activate using the folder name, sometimes